test/common/crudtester.go

```diff
@@ -739,3 +731,25 @@ func (c *FederatedTypeCrudTester) CheckStatusCreated(qualifiedName util.Qualifie
 		c.tl.Fatalf("Timed out waiting for %s %q", statusKind, qualifiedName)
 	}
 }
+
+// GetGenericStatus retrieves a federated resource and converts it to
+// the generic status interface.
+func GetGenericStatus(client genericclient.Client, gvk schema.GroupVersionKind,
+	qualifiedName util.QualifiedName) (*status.GenericFederatedStatus, error) {
+
+	fedObject := &unstructured.Unstructured{}
+	fedObject.SetGroupVersionKind(gvk)
+	err := client.Get(context.TODO(), fedObject, qualifiedName.Namespace, qualifiedName.Name)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to retrieve federated resource from the API")
+	}
+
+	// Convert the resource to the status interface
```

Comment:
Updated to specify struct instead of interface.  The previous use of the term 'interface' was related to  `UnstructuredToInterface`, which unmarshalls to a variable of type `interface{}`.